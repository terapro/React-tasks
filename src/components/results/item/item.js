import React from 'react';
import PropTypes from 'prop-types';
import {ItemPoster} from 'src/components/results/item-poster/item-poster.js';
import {ItemInfo} from 'src/components/results/item-info/item-info.js'

export const Item = ({info, onPosterClick, filmIndex}) => (
    <div className='item'>
        <ItemPoster posterLink={info['poster_path']} posterClick={onPosterClick} filmIndex={filmIndex}/>
        <ItemInfo filmTitle={info['title']} releaseDate={info['release_date']} genre={info['genres']} />
    </div>
);

Item.propTypes = {
  info: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    tagline: PropTypes.string,
    vote_average: PropTypes.number,
    vote_count: PropTypes.number,
    release_date: PropTypes.string,
    poster_path: PropTypes.string,
    overview: PropTypes.string,
    budget: PropTypes.number,
    revenue: PropTypes.number,
    genres: PropTypes.arrayOf(PropTypes.string),
    runtime: PropTypes.number
  }).isRequired,
    onPosterClick: PropTypes.func,
    filmIndex: PropTypes.number.isRequired
};

Item.defaultProps = {
    onPosterClick: f=>f
};