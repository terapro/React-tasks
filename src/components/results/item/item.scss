.item {
  display: inline-block;
  padding: 40px 40px 0 40px;
}