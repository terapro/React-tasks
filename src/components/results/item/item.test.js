import React from 'react';
import {Item} from 'src/components/results/item/item.js';
import {shallow} from 'enzyme';

const mockFunc = jest.fn();

const mockFilm = {
  "id": 333339,
  "title": "Ready Player One",
  "tagline": "A better reality awaits.",
  "vote_average": 8.1,
  "vote_count": 617,
  "release_date": "2018-03-28",
  "poster_path": "https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg",
  "overview": "When the creator of a popular video game system dies, a virtual contest is created to compete for his fortune.",
  "budget": 175000000,
  "revenue": 0,
  "genres": [
    "Adventure",
    "Science Fiction",
    "Action"
  ],
  "runtime": 140
};


describe('<Item />', () => {
  describe('Rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<Item info={mockFilm} />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
