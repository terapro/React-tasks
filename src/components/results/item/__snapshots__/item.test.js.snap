// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> Rendering should render the component 1`] = `
<div
  className="item"
>
  <ItemPoster
    filmIndex={5}
    posterClick={[MockFunction]}
    posterLink="https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg"
  />
  <ItemInfo
    filmTitle="Ready Player One"
    genre={
      Array [
        "Adventure",
        "Science Fiction",
        "Action",
      ]
    }
    releaseDate="2018-03-28"
  />
</div>
`;

exports[`<Item /> Rendering should render the component with default props 1`] = `
<div
  className="item"
>
  <ItemPoster
    filmIndex={5}
    posterClick={[Function]}
    posterLink="https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg"
  />
  <ItemInfo
    filmTitle="Ready Player One"
    genre={
      Array [
        "Adventure",
        "Science Fiction",
        "Action",
      ]
    }
    releaseDate="2018-03-28"
  />
</div>
`;
